src/System.Numerics.Tensors/System/Numerics/DenseTensor.cs

```diff
@@ -98,6 +98,29 @@ public override void SetValue(int index, T value)
             Buffer.Span[index] = value;
         }
 
+        protected override void CopyTo(T[] array, int arrayIndex)
+        {
+            Buffer.Span.CopyTo(array.AsSpan().Slice(arrayIndex));
+        }
+
+        protected override int IndexOf(T item)
+        {
+            // TODO: use Span.IndexOf when/if it removes the struct type constraint
+            if (Buffer.TryGetArray(out var arraySegment))
+            {
+                var result = Array.IndexOf(arraySegment.Array, item, arraySegment.Offset, arraySegment.Count);
+                if (result != -1)
+                {
+                    result -= arraySegment.Offset;
```

Comment:
You also need to ensure that the result is not > array.Length - offset - count

Reply:
I'm not sure how that could happen, can you explain?  I'm calling Array.IndexOf passing in the offset and a count.  Array.IndexOf shouldn't return a result that is past the count, will it?

Reply:
My bad. The line is so long that I did not see the end of it in the default browser view. It's all good.